python/cog/server/runner.py

```diff
@@ -198,8 +198,12 @@ async def inner() -> SetupResult:
             except Exception:
                 logs.append(traceback.format_exc())
                 status = schema.Status.FAILED
+            except asyncio.CancelledError:
+                self.log.info("caught CancelledError during setup")
+                logs.append(traceback.format_exc())
+                status = schema.Status.FAILED
             except BaseException:
```

Comment:
Um, why are we catching `BaseException` and not re-raising it? That's generally a _very_ bad idea.

Reply:
it had appeared that shutdown during setup was failing, in part because this would get cancelled and there would be no setup result. I've removed this